### DISCUSSION TITLE
Email in Flex - HTML Formatting

### DISCUSSION THREAD

**Discussion by Porlin** [twilio-labs/paste#1389] [Office hours] [answered]
**Background:** Email in Flex- Milestone 2 - Features. We are looking at HTML Formatting and I want to get feedback on the icon state usage in the HTML toolbar. (Icons, hover and click state)
**Stage of design:** High-fidelity
**Research:** Include links to any [findings](https://docs.google.com/document/d/1ub_pczr80zTs_c6_JGS3UZTrI_GSIlE581m8Vmlv-1k/edit?usp=sharing)
**Links to mockups or prototype:** [Prototype](https://twilio.invisionapp.com/share/XG10U8Q9UHPE)

**Comment by zahnster** [comment]
@Porlin thank you for this content! Looking forward to having the team discuss this with you at Office Hours this week.

**Comment by serifluous** [comment, marked as answer]
Hey @Porlin ! Thanks for sharing. We'll chat more about the icon toolbar more during office hours.

To your second question about general component usage, all I could spot was:

- In the email drafting container, "CC" is most likely a button, and should use the [link-style button style](url) (semibold, no underline). 
- Re: the "Email sent successfully" message, I know there are historical reasons Flex doesn't use Toasts, but if there's a chance you'd be able to open that discussion again and Email might be able to use [Paste Toasts](https://paste.twilio.design/components/toast)...I figure I'd mention it 🙃

**Comment by nkrantz** [comment]
Hey @Porlin, just following up on today's Office Hours. [Here's the link to the annotated recording](https://www.loom.com/share/ef27bf910d2548aba497a5e85e45b992) (password is in slack). Thanks for coming!

**Comment by Porlin** [comment]
Hey @SiTaggart @serifluous and team,
Just sharing a screen recording based off feedback last week about the 6 button states for the HTML Toolbar. If you have time to take a look and let me know what you think?

https://user-images.githubusercontent.com/49287701/117029064-3705ee80-acf6-11eb-840a-5520e402439c.mov

Porlin

**Reply by nkrantz** [reply]
Hey @Porlin, could you share the button states in a different format? It's a bit hard to tell what's going on in the screen recording. If you could lay them out in screenshots along with labels for which design applies to which state, that would be great.

**Reply by Porlin** [reply]
Hey Nora, 

Sure, I have attached the following png for you to view - please see attached. 
![Email - HTML Toolbar button states example](https://user-images.githubusercontent.com/49287701/117159167-f79fd680-adb7-11eb-92fe-05049c8e3c05.png)

**Reply by SiTaggart** [reply]
Thanks for this, it makes it much easier to see whats's going on. I think it's super close, but not quite there yet. The fact its so close to our new Pagination design is a happy coincidence and oddly enough has highlighted issues with our designs, [found here](https://paste.twilio.design/components/pagination#pagination-with-total-page-count). So cc @katieporter for when you handle the dark mode selected state updates, you'll need to consider this too.

The states we really need to nail are:

- Default
- Default hovered
- Default focused
- Pressed
- Pressed hovered
- Pressed focused
- Default active (optional)
- Pressed active (optional)

I think you potentially have 5 covered, and unless you feel that active is essential to communicating the interaction I think you can leave those to be the same as hover, meaning you can use something like the pressed off as an extra one to cover the gaps.

It's mainly more apparent here because you can have more than one button pressed at the same time. So making it visually different between pressed and pressed and hovered is pretty important.

**Reply by Porlin** [reply]
Thanks for this feedback @SiTaggart These names make sense. Looks like it’ll cover all of the states we need.
Sarah/Katie have shared their figma file with me for the dark mode and Pagination. I'll have a play around with it too.
